lib/node/http.js

```diff
@@ -452,6 +475,79 @@ class HTTP extends Server {
 
       res.json(200, { success: true });
     });
+
+    // Initiate a zone dump to S3
+    this.post('/dump-zone-to-s3', async (req, res) => {
+      if(this.s3 === null) {
+        res.json(501,
+          {
+            success: false,
+            message: 'AWS is not properly configured'
+          }
+        );
+        return;
+      }
+
+      if(this.runningUpload !== null) {
+        res.json(202,
+          {
+            success: true,
+            message: 'upload already in progress'
+          }
+        );
+        return;
+      }
+
+      this.runningUpload = this.s3.upload({
+        Bucket: this.options.s3DumpConfig.bucket,
+        Key: this.options.s3DumpConfig.key,
+        Body: dumpzone.readableStream(this.chain)
+      }, (err, data) => {
+        // TODO - capture status, do a rename?
```

Comment:
Not blocking/not requesting this but might be nice for the key to have the timestamp and then to rename one "current" etc

Reply:
:+1: I'll verify how rename works in S3

Reply:
There's actually no way to rename something in S3, would have to do a copy then delete. Alternative could be to push files with the timestamp and have some other process to reap ones older than a certain age, but it would require the consumer to list the objects in the bucket and chose the lastest one

Reply:
one other alternative might be enabling versioning on s3 bucket by default you will get the timestamp and maangement of old versions provided by aws

Reply:
Versioning sounds good :+1: